equinox/nn/pool.py

```diff
@@ -85,6 +91,26 @@ def __init__(
                 f"{num_spatial_dims} containing ints or tuples of length 2."
             )
 
+        for padding, kernel_size in zip(self.padding, self.kernel_size):
+            if kernel_size / 2 < padding[0] or kernel_size / 2 < padding[1]:
+                raise ValueError(
+                    "Padding should be less than or equal to half of the kernel size."
+                    f" Received kernel_size {kernel_size} and padding {padding}."
+                )
+
+    def _update_padding_for_ceil(self, input_shape):
+        new_padding = []
+        total_padding = np.sum(self.padding, axis=1)
+        for idx, (input_dim, padding, kernel_size, stride) in enumerate(
+            zip(input_shape[1:], total_padding, self.kernel_size, self.stride)
+        ):
+            target_out_dim = np.ceil(
+                (input_dim + padding - 1 * (kernel_size - 1) - 1) / stride + 1
+            ).item()
+            padding = int((target_out_dim - 1) * stride + kernel_size - input_dim)
+            new_padding.append((self.padding[idx][0], padding - self.padding[idx][0]))
+        return tuple(new_padding)
```

Comment:
This feels overly-complicated. If `k % n == 0` then `floor(n/k) == ceil(n/k)`, otherwise `floor(n/k) + 1 == ceil(n/k)`.

Thus I think something like this should suffice:
```python
def _update_padding_for_ceil(self, input_shape):
  new_padding = []
  for input_size, (left_padding, right_padding), kernel_size, stride in zip(
                                    input_shape, self.padding, self.kernel_size, self.stride):
    if stride % (input_size + left_padding + right_padding - kernel_size + 1) == 0:
      # the sliding kernel exactly matches up with the edge of the input, with no gaps,
      # if there exists x such that
      # x * stride + kernel_size -1 == input_size + left_padding + right_padding
      new_padding.append((left_padding, right_padding))
    else:
      new_padding.append((left_padding, right_padding + stride))
  return tuple(new_padding)
```
I think?

Reply:
> This feels overly-complicated. If k % n == 0 then floor(n/k) == ceil(n/k), otherwise floor(n/k) + 1 == ceil(n/k)

The solution was also based on it, but yeah yours does look simpler.

Reply:
Wanted to cross-check some corrections to your solutions. 

- A minor correction: `input_shape` in `for` should be `input_shape[1:]` but can be ignored as it can be handled by calling the function with the required values
- `stride % (input_size + left_padding + right_padding - kernel_size + 1)` should be flipped around and `1` should be dropped. This is based on the formulae mentioned in [PyTorch](https://pytorch.org/docs/stable/generated/torch.nn.MaxPool1d.html?highlight=maxpool#torch.nn.MaxPool1d). 
- Shouldn't the new padding to be added be something like `right_padding + stride - (remainder from the % operation)`?

Reply:
- Agreed
- Would need to think about the `1`; otherwise agreed;
- I don't think it matters. We don't need to perfectly align the sliding window with the boundary of the input; we just need to add enough padding that we take one more step.

Reply:
3) Agreed, but adding additional padding than required can potentially output `identity` values in the output as per the discussion above.
I'm in favour of suppressing the padding and also matching the correct padding `right_padding + stride - (remainder from the % operation)` . Thoughts?

Reply:
So I'm saying that I don't think "the remainder from the % operation" shouldn't affect things.

By adding on `stride` extra padding here, we should obtain precisely one more step, of length `stride`, than before?

Reply:
I was thinking of a scenario when `kernel_size < right_padding + stride` as many times `kernel_size = stride` is used. 
However, I think the failure will happen regardless of using `stride` or `remainder`. This will happen when with `left_padding`, the `kernel` and `stride` exactly covers the input but `right _padding + _pad > kernel_size`. We will have identity values introduced into the computation. 

Example: 
```
x = jnp.arange(14).reshape(1, 14)
max_pool = eqx.nn.MaxPool1d(kernel_size=3, stride=3, padding=1, use_ceil=True)
max_pool(x)  # [[ 1, 4, 7, 10, 13, -2147483648 ]]
```

Reply:
Maybe we can simply use `+stride` and simply raise a warning at runtime with a check to let the user know that `identity` have crept in with `ceil`?

Reply:
Perhaps let's move the current is-padding-sensible check to after `use_ceil` is resolved here, and raise a `RuntimeError` if it fails.